service/builder/pipelines_builder.go

```diff
@@ -113,6 +116,11 @@ func (pb *PipelinesBuilder) buildPipeline(
 		case configmodels.TracesDataType:
 			var proc processor.TraceProcessor
 			proc, err = factory.CreateTraceProcessor(pb.logger, tc, procCfg)
+			if procName == "tail_sampling" {
```

Comment:
We definitely don't want special cases like this in pipeline builder which is universal and has no special code path for any other component.